react/ModalContext.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { createContext } from 'react'
+
+const { Consumer, Provider } = createContext({})
+
+interface ModalContext {
+  closeModal: () => void
+}
+
+const modalConsumer = <P extends object>(
```

Comment:
```suggestion
const withModal = <P extends object>(
``` 
may be more consistent